File: src/main/java/back/ecommerce/admin/dto/response/RequestProductDto.java
package back.ecommerce.admin.dto.response;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;

import back.ecommerce.product.entity.ApprovalStatus;
import back.ecommerce.product.entity.Category;
import back.ecommerce.product.entity.RequestProduct;
import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Builder()
@Getter
public class RequestProductDto {
	private final Long requestId;
	private final String email;
	private final String name;
	private final String brandName;
	private final Category category;
	private final Long price;
	private final ApprovalStatus approvalStatus;
	@JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
	private final LocalDateTime requestTime;

	public static RequestProductDto create(RequestProduct requestProduct) {
		return RequestProductDto.builder()
			.requestId(requestProduct.getId())
			.email(requestProduct.getEmail())
			.name(requestProduct.getName())
			.brandName(requestProduct.getBrandName())
			.category(requestProduct.getCategory())
			.price(requestProduct.getPrice())
			.approvalStatus(requestProduct.getApprovalStatus())
			.requestTime(requestProduct.getCreatedDate())
			.build();
	}
}

File: src/main/java/back/ecommerce/admin/dto/response/RequestProductDto.kt
package back.ecommerce.admin.dto.response

import back.ecommerce.product.entity.ApprovalStatus
import back.ecommerce.product.entity.Category
import back.ecommerce.product.entity.RequestProduct
import com.fasterxml.jackson.annotation.JsonFormat
import java.time.LocalDateTime

data class RequestProductDto(
    val requestId: Long,
    val email: String,
    val name: String,
    val brandName: String,
    val category: Category,
    val price: Long,
    val approvalStatus: ApprovalStatus,
    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val requestTime: LocalDateTime
) {
    constructor(requestProduct: RequestProduct) : this(
        requestProduct.id,
        requestProduct.email,
        requestProduct.name,
        requestProduct.brandName,
        requestProduct.category,
        requestProduct.price,
        requestProduct.approvalStatus,
        requestProduct.createdDate,
    )

    companion object {
        fun create(requestProduct: RequestProduct) = RequestProductDto(requestProduct)
    }
}